src/logging.rs

```diff
@@ -59,9 +58,7 @@ pub fn setup_layers(
         .join(","),
     )?;
 
-    let kvp_enabled = config
-        .map(|cfg| cfg.telemetry.kvp_diagnostics)
-        .unwrap_or(true);
+    let kvp_enabled = config.telemetry.kvp_diagnostics;
```

Comment:
It's fine as it is, but since you don't have a fallback anymore you can just use `if config.telemetry.kvp_diagnostics {` directly